enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,574 @@
+---
+
+
+# Index representation and management 
+
+### Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+### Summary 
+
+This enhancement proposes the representation of packages in an index in a standard declarative way, so that index authors and operator authors can communicate with each other about the content of an index using json/yaml, without needing to communicate with each other using artifacts(operator manifests etc). Using the declarative representation of the packages in the index, it also proposes allowing index authors and individual package authors to alter the content of the packages(package name, channel information etc) as well as the structure of the packages (channel upgrade graphs etc) of the index without having to rebuild bundles+republish the index.
+
+### Motivation
+
+In the current state of the world, operators are introduced to operator-lifecycle-manager by representing them as [Operator Bundles](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#operator-bundle). These operator bundles are then loaded onto a a container using the [opm](https://github.com/operator-framework/operator-registry/blob/master/docs/design/opm-tooling.md) tool (`opm index add --bundles`) to build an index of operators. Operator bundles conceptually belong to a channel in a package, with each bundle having the ability to upgrade to a different bundle in the channel. 
+
+![Alt text](assets/community-operators.png?raw=true "community-operators")
+
+The way we describe the intent of a bundle to be on a particular channel in a package is by hardcoding that information in the [bundle annotations](https://github.com/operator-framework/operator-registry/blob/release-4.6/docs/design/operator-bundle.md#bundle-annotations) inside the bundle metadata itself. However, the channel that the bundle wants to be in is more of a package level information. Hardcoded channel information inside a bundle restricts a bundle in a particular channel once the bundle is built and loaded into an index. This makes it difficult to change channel info retroactively. When a bundle is added to an index using `opm`, the `mode` (one of `replaces`|`semver`|`semver-skippatch`) in which the bundle is added (`opm index add --bundles <list-of-bundles> --mode <replaces|semver|semver-skippatch>`) determines the upgrade graph of bundles in the channel (which bundle can be upgrade to which). This upgrade graph for bundles in a channel can also not be changed retroactively. The only way to change the channel info, the upgrade graph or any package level information currently is to rebuild the bundles with the modified package information and then republish the index. This is done by making additive changes to the index using the`opm` tool. 
+
```

Comment:
Who does that and when this process is required? I mean what are the use case/scenarios that make this action required. Could you provide an example? I think it would clarifies a lot.